src/include/mpi.h.in

```diff
@@ -506,6 +506,7 @@ extern int * const MPI_WEIGHTS_EMPTY MPICH_API_PUBLIC;
 
 #define MPI_BOTTOM      (void *)0
 #define MPI_IN_PLACE  (void *) -1
+#define MPI_BUFFER_AUTOMATIC (void *)0
```

Comment:
does this need additional support in the Fortran bindings?

Reply:
Good point

Reply:
Commits added.

Reply:
@hzhou A value of `0` is very to missuse/confuse with `NULL`. If you keep this definition as it is now, then `MPI_Buffer_attach(MPI_BUFFER_AUTOMATIC, ...)` would be equivalent to `MPI_Buffer_attach(NULL, ...)`. The first is is a clear intent to enable automatic buffering, the second feels like a bug. Yet, you cannot distinguish between them to generate an error. Why don't you just use let say `(void *) -2` ?.

Reply:
I guess I want to prove to myself that our implementation is ABI-neutral :) Since there is preference, let's make it `(void *) -2`. I'll add as an additional commit.